domain/src/test/java/org/oppia/domain/audio/AudioPlayerControllerTest.kt

```diff
@@ -0,0 +1,252 @@
+  }
+
+  @Test
+  fun testAudioPlayer_pause_notIsPlaying() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.pause()
+
+    assertThat(audioPlayerController.isPlaying()).isFalse()
+    assertThat(shadowMediaPlayer.isReallyPlaying).isFalse()
+  }
+
+  @Test
+  fun testAudioPlayer_seekTo_hasCorrectProgress() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.seekTo(500)
+
+    assertThat(shadowMediaPlayer.currentPositionRaw).isEqualTo(500)
+  }
+
+  @Test
+  fun testAudioPlayer_releaseMediaPlayer_isReleasedState() {
+    arrangeMediaPlayer()
+
+    audioPlayerController.releaseMediaPlayer()
+
+    assertThat(shadowMediaPlayer.state).isEqualTo(ShadowMediaPlayer.State.END)
+  }
+
+  @Test
+  fun testAudioObserver_invokePrepare_capturesPreparedState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+
+    arrangeMediaPlayer()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.PREPARED)
+  }
+
+  @Test
+  fun testAudioObserver_invokeCompletion_capturesCompletedState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    shadowMediaPlayer.invokeCompletionListener()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.COMPLETED)
+  }
+
+  @Test
+  fun testAudioObserver_invokeChangeDataSource_capturesPendingState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    audioPlayerController.changeDataSource(TEST_URL)
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isPending()).isTrue()
+  }
+
+  @Test
+  fun testAudioObserver_invokePlay_capturesPlayingState() {
+    audioPlayerController.getPlayState().observeForever(mockAudioPlayerObserver)
+    arrangeMediaPlayer()
+
+    audioPlayerController.play()
+    audioPlayerController.pause()
+
+    verify(mockAudioPlayerObserver, atLeastOnce()).onChanged(audioPlayerResultCaptor.capture())
+    assertThat(audioPlayerResultCaptor.value.isSuccess()).isTrue()
+    assertThat(audioPlayerResultCaptor.value.getOrThrow().type).isEqualTo(AudioPlayerController.PlayStatus.PAUSED)
+  }
+
+  private fun arrangeMediaPlayer() {
+    audioPlayerController.initializeMediaPlayer(TEST_URL)
+    shadowMediaPlayer.invokePreparedListener()
+  }
+
+  private fun addMediaInfo() {
+    val dataSource = DataSource.toDataSource(context , Uri.parse(TEST_URL))
+    val mediaInfo = ShadowMediaPlayer.MediaInfo(1000, 0)
```

Comment:
Please add in-line comments with argument names to provide context for these literals, e.g.:

```kotlin
ShadowMediaPlayer.MediaInfo(/* firstArgName= */ 1000, /* secondArgName= */ 0)
```

This helps provide context to readers that don't have an IDE to see the argument name.

Reply:
updated